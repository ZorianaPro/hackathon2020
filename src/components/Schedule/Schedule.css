.Schedule {
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 100vh;
    flex-wrap: wrap;
    position: relative;
    flex-direction: column;
    padding: 4.5em 0;
}

.Schedule-Wrapper {
    display: flex;
    align-items: flex-start;
    justify-content: center;
}

.Day {
    flex: 0 1 400px;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    margin: 0 2em 0;
}

.Lightnings-Left,
.Lightnings-Right {
    position: absolute;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    height: 100%;
    z-index: -1;
}

.Lightnings-Left {
    left: 0;
}

.Lightnings-Right {
    right: 0;
}

@media only screen and (max-width: 992px) {
  .Schedule-Wrapper {
    flex-direction: column;
    margin: 0 3em 0;
  }

  .Day {
    width: 100%;
    margin: 0;
  }
}


@media only screen and (max-width: 992px) {
  .Lightnings-Left,
  .Lightnings-Right {
        display: none;
    }
}

.Schedule table {
    width: 100%;
}

.Schedule table tr:nth-child(even) th {
    color: #ccc;
    font-weight: normal;
}
.Schedule table th, .Schedule table td {
    padding: 0.5rem 1rem;
}
.Schedule table th {
    font-weight: normal;
    border-top: thin dotted #ccc;
}
.Schedule table td {
    font-size: var(--fontSize16);
    border-radius: 9px;
    transition: opacity 0.3s ease;
}
.Schedule table td > span {
    display: block;
    width: 100%;
}

.is-hacking {
 background-color: var(--colorRadicalRed);
 color: white;
}

.is-official-program {
  background-color: var(--colorSpaceCadet);
  color: white;
}

.Schedule-Title {
    font-size: var(--fontSize56);
    font-weight: var(--fontWeightPoppinsBold);
    font-style: italic;
    line-height: var(--lineHeight);
    text-align: center;
    color: var(--colorSpaceCadet);
}

.Schedule .Lightnings-Left .LightningSVG:nth-of-type(1) {
    transform: scale(0.5) translateX(50px);
}

.Schedule .Lightnings-Left .LightningSVG:nth-of-type(2) {
    transform: scale(0.3);
}

.Schedule .Lightnings-Left .LightningSVG:nth-of-type(3) {
    transform: scale(1);
}

.Schedule .Lightnings-Left .LightningSVG:nth-of-type(4) {
    transform: scale(0.4);
}

.Schedule .Lightnings-Left .LightningSVG:nth-of-type(5) {
    transform: scale(0.8);
}

.Schedule .Lightnings-Left .LightningSVG:nth-of-type(6) {
    transform: scale(0.6);
}

/* Right */
.Schedule .Lightnings-Right .LightningSVG:nth-of-type(1) {
    transform: scale(0.5) translateX(-50px);
}

.Schedule .Lightnings-Right .LightningSVG:nth-of-type(2) {
    transform: scale(0.3);
}

.Schedule .Lightnings-Right .LightningSVG:nth-of-type(3) {
    transform: scale(1);
}

.Schedule .Lightnings-Right .LightningSVG:nth-of-type(4) {
    transform: scale(0.4);
}

.Schedule .Lightnings-Right .LightningSVG:nth-of-type(5) {
    transform: scale(0.8);
}

.Schedule .Lightnings-Right .LightningSVG:nth-of-type(6) {
    transform: scale(0.6);
}