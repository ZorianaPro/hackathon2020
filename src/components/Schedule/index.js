import React from "react";
import "./Schedule.css";

const Schedule = ({ }) => {
  return (
    <div className="Schedule">
      <p className="Schedule-Title">Schedule</p>
      <div className="Schedule-Wrapper">
        <div className="Day">
          <h2>Day 1</h2>
          <table>
            <tbody>
              <tr>
                <th>09:00</th>
                <td colSpan="4" rowSpan="4" className="is-official-program">
                  Official welcome to Official Day 1{" "}
                  <span>online (link TBA)</span>
                </td>
              </tr>
              <tr>
                <th>09:30</th>
              </tr>
              <tr>
                <th>10:00</th>
              </tr>
              <tr>
                <th>10:30</th>
              </tr>
              <tr>
                <th>11:00</th>
                <td rowSpan="6" className="is-hacking">
                  Hack, hack, hack
              </td>
              </tr>
              <tr>
                <th>11:30</th>
              </tr>
              <tr>
                <th>12:00</th>
              </tr>
              <tr>
                <th>12:30</th>
              </tr>
              <tr>
                <th>13:00</th>
              </tr>
              <tr>
                <th>13:30</th>
              </tr>
              <tr>
                <th>14:00</th>
                <td colSpan="4" rowSpan="2" className="is-official-program">
                  Virtual coffee & check in <span>online (link TBA)</span>
                </td>
              </tr>
              <tr>
                <th>14:30</th>
              </tr>
              <tr>
                <th>15:00</th>
                <td rowSpan="2" className="is-hacking">
                  Hack, hack, hack
              </td>
              </tr>
              <tr>
                <th>15:30</th>
              </tr>
              <tr>
                <th>16:00</th>
                <td colSpan="4" rowSpan="3" className="is-official-program">
                  Virtual yoga <span>online (link TBA)</span>
                </td>
              </tr>
              <tr>
                <th>16:15</th>
              </tr>
              <tr>
                <th>16:30</th>
              </tr>
              <tr>
                <th>17:00</th>
                <td colSpan="4" rowSpan="8" className="is-hacking">
                  Hack, hack, hack
              </td>
              </tr>
              <tr>
                <th>17:15</th>
              </tr>
              <tr>
                <th>17:30</th>
              </tr>
              <tr>
                <th>18:00</th>
              </tr>
              <tr>
                <th>18:30</th>
              </tr>
              <tr>
                <th>19:00</th>
              </tr>
            </tbody>
          </table>
        </div>
        <div className="Day">
          <h2>Day 2</h2>
          <table>
            <tbody>
              <tr>
                <th>09:00</th>
                <td colSpan="4" rowSpan="2" className="is-official-program">
                  Welcome to Day 2 <span>online (link TBA)</span>
                </td>
              </tr>
              <tr>
                <th>09:30</th>
              </tr>
              <tr>
                <th>10:00</th>
                <td rowSpan="12" className="is-hacking">
                  Hack, hack, hack
              </td>
              </tr>
              <tr>
                <th>10:30</th>
              </tr>
              <tr>
                <th>11:00</th>
              </tr>
              <tr>
                <th>11:30</th>
              </tr>
              <tr>
                <th>12:00</th>
              </tr>
              <tr>
                <th>12:30</th>
              </tr>
              <tr>
                <th>13:00</th>
              </tr>
              <tr>
                <th>13:30</th>
              </tr>
              <tr>
                <th>14:00</th>
              </tr>
              <tr>
                <th>14:30</th>
              </tr>
              <tr>
                <th>15:00</th>
              </tr>
              <tr>
                <th>15:30</th>
              </tr>
              <tr>
                <th>16:00</th>
                <td colSpan="4" rowSpan="1" className="is-official-program">
                  Stop coding!
              </td>
              </tr>
              <tr>
                <th>16:15</th>
                <td colSpan="4" rowSpan="2" className="is-official-program">
                  Demos
              </td>
              </tr>
              <tr>
                <th>16:30</th>
              </tr>
              <tr>
                <th>17:00</th>
                <td colSpan="4" rowSpan="1" className="is-official-program">
                  Voting <span>online (link tba)</span>
                </td>
              </tr>
              <tr>
                <th>17:15</th>
                <td colSpan="4" rowSpan="2" className="is-official-program">
                  Announce winners & ending ceremony{" "}
                  <span>online (link tba)</span>
                </td>
              </tr>
              <tr>
                <th>17:30</th>
              </tr>
              <tr>
                <th>18:00</th>
              </tr>
              <tr>
                <th>18:30</th>
              </tr>
              <tr>
                <th>19:00</th>
                <td colSpan="4" rowSpan="2" className="is-official-program">
                  Offline afterparty for those who want to join{" "}
                  <span>Biergarten (tba)</span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>

    </div>
  );
};

export default Schedule;
