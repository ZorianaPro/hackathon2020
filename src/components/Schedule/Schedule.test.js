import React from 'react';
import { shallow, mount } from 'enzyme';
import Schedule from '../Schedule';

describe('Schedule', () => {
  let schedule;

  beforeEach(() => {
    schedule = mount(<Schedule />);
  });

  it('renders without crashing', () => {
    expect(shallow.bind(shallow, <Schedule />)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(schedule).toMatchSnapshot();
  });
});