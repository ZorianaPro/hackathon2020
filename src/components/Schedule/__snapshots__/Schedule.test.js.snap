// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule renders as expected 1`] = `
<Schedule>
  <div
    className="Schedule"
    id="schedule"
  >
    <p
      className="Schedule-Title"
    >
      Schedule
    </p>
    <div
      className="Lightnings-Left"
    >
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
    </div>
    <div
      className="Schedule-Wrapper"
    >
      <div
        className="Day"
      >
        <h2>
          Day 1
        </h2>
        <table>
          <tbody>
            <tr>
              <th>
                09:00
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="4"
              >
                Official welcome to Official Day 1
                 
                <span>
                  online (link TBA)
                </span>
              </td>
            </tr>
            <tr>
              <th>
                09:30
              </th>
            </tr>
            <tr>
              <th>
                10:00
              </th>
            </tr>
            <tr>
              <th>
                10:30
              </th>
            </tr>
            <tr>
              <th>
                11:00
              </th>
              <td
                className="is-hacking"
                rowSpan="6"
              >
                Hack, hack, hack
              </td>
            </tr>
            <tr>
              <th>
                11:30
              </th>
            </tr>
            <tr>
              <th>
                12:00
              </th>
            </tr>
            <tr>
              <th>
                12:30
              </th>
            </tr>
            <tr>
              <th>
                13:00
              </th>
            </tr>
            <tr>
              <th>
                13:30
              </th>
            </tr>
            <tr>
              <th>
                14:00
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="2"
              >
                Virtual coffee & check in 
                <span>
                  online (link TBA)
                </span>
              </td>
            </tr>
            <tr>
              <th>
                14:30
              </th>
            </tr>
            <tr>
              <th>
                15:00
              </th>
              <td
                className="is-hacking"
                rowSpan="2"
              >
                Hack, hack, hack
              </td>
            </tr>
            <tr>
              <th>
                15:30
              </th>
            </tr>
            <tr>
              <th>
                16:00
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="3"
              >
                Virtual yoga 
                <span>
                  online (link TBA)
                </span>
              </td>
            </tr>
            <tr>
              <th>
                16:15
              </th>
            </tr>
            <tr>
              <th>
                16:30
              </th>
            </tr>
            <tr>
              <th>
                17:00
              </th>
              <td
                className="is-hacking"
                colSpan="4"
                rowSpan="8"
              >
                Hack, hack, hack
              </td>
            </tr>
            <tr>
              <th>
                17:15
              </th>
            </tr>
            <tr>
              <th>
                17:30
              </th>
            </tr>
            <tr>
              <th>
                18:00
              </th>
            </tr>
            <tr>
              <th>
                18:30
              </th>
            </tr>
            <tr>
              <th>
                19:00
              </th>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="Day"
      >
        <h2>
          Day 2
        </h2>
        <table>
          <tbody>
            <tr>
              <th>
                09:00
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="2"
              >
                Welcome to Day 2 
                <span>
                  online (link TBA)
                </span>
              </td>
            </tr>
            <tr>
              <th>
                09:30
              </th>
            </tr>
            <tr>
              <th>
                10:00
              </th>
              <td
                className="is-hacking"
                rowSpan="12"
              >
                Hack, hack, hack
              </td>
            </tr>
            <tr>
              <th>
                10:30
              </th>
            </tr>
            <tr>
              <th>
                11:00
              </th>
            </tr>
            <tr>
              <th>
                11:30
              </th>
            </tr>
            <tr>
              <th>
                12:00
              </th>
            </tr>
            <tr>
              <th>
                12:30
              </th>
            </tr>
            <tr>
              <th>
                13:00
              </th>
            </tr>
            <tr>
              <th>
                13:30
              </th>
            </tr>
            <tr>
              <th>
                14:00
              </th>
            </tr>
            <tr>
              <th>
                14:30
              </th>
            </tr>
            <tr>
              <th>
                15:00
              </th>
            </tr>
            <tr>
              <th>
                15:30
              </th>
            </tr>
            <tr>
              <th>
                16:00
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="1"
              >
                Stop coding!
              </td>
            </tr>
            <tr>
              <th>
                16:15
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="2"
              >
                Demos
              </td>
            </tr>
            <tr>
              <th>
                16:30
              </th>
            </tr>
            <tr>
              <th>
                17:00
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="1"
              >
                Voting 
                <span>
                  online (link tba)
                </span>
              </td>
            </tr>
            <tr>
              <th>
                17:15
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="4"
              >
                Announce winners & ending ceremony
                 
                <span>
                  online (link tba)
                </span>
              </td>
            </tr>
            <tr>
              <th>
                17:30
              </th>
            </tr>
            <tr>
              <th>
                18:00
              </th>
            </tr>
            <tr>
              <th>
                18:30
              </th>
            </tr>
            <tr>
              <th>
                19:00
              </th>
              <td
                className="is-official-program"
                colSpan="4"
                rowSpan="2"
              >
                Offline afterparty for those who want to join
                 
                <span>
                  Biergarten (tba)
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      className="Lightnings-Right"
    >
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
      <LightningSVG>
        <svg
          className="LightningSVG"
        >
          <use
            xlinkHref="#LightningSymbol"
          />
        </svg>
      </LightningSVG>
    </div>
  </div>
</Schedule>
`;
